modular_skyrat/modules/container_emotes/code/container_emotes.dm

```diff
@@ -26,15 +27,31 @@
 	else if(user.client?.prefs?.muted & MUTE_IC)
 		tgui_alert(user, "You cannot send IC messages (muted).")
 		return FALSE
-	else if (isturf(user.loc))
-		to_chat(user, "You are not within anything!") // If user is banned from chat, emotes, or the user is not within anything (ex. a locker) return.
+
+	var/times_searched = 0
+	var/can_use = TRUE
+	var/atom/current_loc = user.loc
+	var/list/locs_we_can_use = list()
+	if (current_loc != null && !isturf(current_loc))
+		locs_we_can_use += current_loc
+		while (times_searched++ < EXME_MAX_LOC_RECURSION)
+			var/atom/recursive_loc = current_loc.loc
+			if (recursive_loc == null || (isarea(recursive_loc))) // if youre in something already, its fair to say you might be in, say, a pipe. you cant use that for emoting, so the floor will have to do
+				break
+			locs_we_can_use += recursive_loc
+			current_loc = recursive_loc
```

Comment:
Does it achieve the same results? Yes. Is it less confusing to look at for someone that isn't the most familiar with coding? Also yes. This is far from being hot enough to really warrant this kind of micro-optimization.
```suggestion

		while (times_searched < EXME_MAX_LOC_RECURSION)
			times_searched++
			var/atom/recursive_loc = current_loc.loc

			if (recursive_loc == null || (isarea(recursive_loc))) // if youre in something already, its fair to say you might be in, say, a pipe. you cant use that for emoting, so the floor will have to do
				break

			locs_we_can_use += recursive_loc
			current_loc = recursive_loc

```

Reply:
fine by me

Reply:
as in, your change is fine